src/main/java/erika/Erika.java

```diff
@@ -0,0 +1,293 @@
+package erika;
+
+import erika.exception.*;
+import erika.filesystem.FileSystem;
+import erika.settings.Settings;
+import erika.task.Deadline;
+import erika.task.Event;
+import erika.task.Task;
+import erika.task.Todo;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+
+public class Erika {
+    private static ArrayList<Task> tasks = new ArrayList<>();
+    private static Scanner in = new Scanner(System.in);
+    private static int markIndex;
+    static FileSystem fileSystem = new FileSystem(Settings.FILENAME, Settings.SEPARATOR);
+
+    public static void main(String[] args) {
+        printWelcomeMessage();
+        initializeFileSystem();
+
+        while (true) {
+            if (mainLoop()) break;
+        }
+    }
+
+    private static void initializeFileSystem() {
+        try{
+            tasks = fileSystem.readFromFile();
+        } catch (FileNotFoundException e){
+            fileSystem.printFileNotFoundMessage();
+        } catch (FileFormatErrorException e) {
+            reinitFileSystem();
+        } catch (IOException e) {
+            printMessage("IO Error encountered when setting up fileSystem");
+        }
+    }
+
+    private static void reinitFileSystem() {
+        try{
+            printMessage(Settings.FILENAME + " is corrupted, re-creating the file");
+            fileSystem.writeToFile("",false);
+        } catch (IOException _e) {
+            printMessage("IO Error encountered when re-creating " + Settings.FILENAME);
+        }
+    }
+
+    private static boolean mainLoop() {
+        try{
+            String line = collectUserInput();
+            if (line.equals("bye")) {
+                printGoodbyeMessage();
+                return true;
+            } else if (line.equals("list")) {
+                printList();
+            } else if (line.contains("mark")) {
+                indexOperation(line, false);
+            } else if (line.contains("todo")) {
+                addTodo(line);
+            } else if (line.contains("deadline")) {
+                addDeadline(line);
+            } else if (line.contains("event")) {
+                addEvent(line);
+            } else if (line.contains("delete")) {
+                indexOperation(line, true);
+            } else {
+                throw new UnknownCommandException();
+            }

```

Comment:
Consider using switch case for clearer code organization and because the values expected are predetermined.